From b093a84ec2ce5ce030f305ef04a741259b5e1f6d Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Mon, 16 Jul 2012 14:27:03 +0800
Subject: [PATCH 12/23] powerpc/fsl: add the usb device a little later

The swiotlb_setup_bus_notifier is registered via arch_initcall, and
the fsl usb device nodes is also added with arch_initcall. So the
usb device may be registered before the swiotlb bus notifier, and
then it will be not set to the correct dma mask. So add this device
a little later.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/sysdev/fsl_soc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/sysdev/fsl_soc.c b/arch/powerpc/sysdev/fsl_soc.c
index 7afe06e..02b1d05 100644
--- a/arch/powerpc/sysdev/fsl_soc.c
+++ b/arch/powerpc/sysdev/fsl_soc.c
@@ -395,7 +395,7 @@ err:
 	return ret;
 }
 
-arch_initcall(fsl_usb_of_init);
+arch_initcall_sync(fsl_usb_of_init);
 
 #if defined(CONFIG_FSL_SOC_BOOKE) || defined(CONFIG_PPC_86xx)
 static __be32 __iomem *rstcr;
-- 
1.7.0

